forum/users/models.py

```diff
@@ -1,3 +1,65 @@
 from django.db import models
 
-# Create your models here.
+from django.db import models
+from django.contrib.auth.models import (
+    AbstractBaseUser,
+    BaseUserManager,
+    PermissionsMixin,
+)
+import datetime
+import re
+
+ROLE_CHOICES = [
+    (0, "Unassigned"),
+    (1, "Startup"),
+    (2, "Investor"),
+]
+
+
+class CustomUserManager(BaseUserManager):
+    def create_user(self, email, password=None, **extra_fields):
+        if email:
+            email_pattern = r"^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[a-zA-Z0-9]+$"
+            if not re.match(email_pattern, email):
+                raise ValueError("The Email is not valid")
+        else:
+            raise ValueError("The Email field must be set")
+        if password:
+            password_pattern = r"^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z]).{8,}$"
+            if not re.match(password_pattern, password):
+                raise ValueError("Password is not matching valid pattern")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password=None, **extra_fields):
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+        return self.create_user(email, password, **extra_fields)
+
+
+class CustomUser(AbstractBaseUser, PermissionsMixin):
+    user_id = models.AutoField(primary_key=True)
+    first_name = models.CharField(max_length=30)
+    last_name = models.CharField(max_length=30)
+    email = models.EmailField(unique=True)
+    password = models.CharField(max_length=128)
```

Comment:
The `password` field is explicitly defined, which may cause redundancy with `AbstractBaseUser`.
It'll be better to remove the `password` field from the model since `AbstractBaseUser` already provides it. Instead, use `set_password` to handle hashing securely.